pallets/tokens/src/lib.rs

```diff
@@ -875,3 +911,256 @@ where
+		let actual = Module::<T>::slash_reserved(currency_id, who, value);
+		(MultiTokenNegativeImbalance::zero(currency_id), actual)
+	}
+
+	fn reserved_balance(currency_id: T::CurrencyId, who: &T::AccountId) -> Self::Balance {
+		Module::<T>::reserved_balance(currency_id, who)
+	}
+
+	fn reserve(currency_id: T::CurrencyId, who: &T::AccountId, value: Self::Balance) -> DispatchResult {
+		Module::<T>::reserve(currency_id, who, value)
+	}
+
+	fn unreserve(currency_id: T::CurrencyId, who: &T::AccountId, value: Self::Balance) -> Self::Balance {
+		Module::<T>::unreserve(currency_id, who, value)
+	}
+
+	fn repatriate_reserved(
+        currency_id: T::CurrencyId,
+		slashed: &T::AccountId,
+		beneficiary: &T::AccountId,
+		value: Self::Balance,
+		status: Status,
+	) -> result::Result<Self::Balance, DispatchError> {
+		Module::<T>::repatriate_reserved(currency_id, slashed, beneficiary, value, status)
+	}
+}
+
+impl<T> MultiTokenLockableCurrency<T::AccountId> for MultiTokenCurrencyAdapter<T>
+where
+	T: Trait,
+{
+	type Moment = T::BlockNumber;
+	type MaxLocks = ();
+
+	fn set_lock(currency_id: T::CurrencyId, id: LockIdentifier, who: &T::AccountId, amount: Self::Balance, _reasons: WithdrawReasons) {
+		Module::<T>::set_lock(id, currency_id, who, amount)
+	}
+
+	fn extend_lock(currency_id: T::CurrencyId, id: LockIdentifier, who: &T::AccountId, amount: Self::Balance, _reasons: WithdrawReasons) {
+		Module::<T>::extend_lock(id, currency_id, who, amount)
+	}
+
+	fn remove_lock(currency_id: T::CurrencyId, id: LockIdentifier, who: &T::AccountId) {
+		Module::<T>::remove_lock(id, currency_id, who)
+	}
+}
+
+
+impl<T> MultiTokenCreatableCurrency<T::AccountId> for MultiTokenCurrencyAdapter<T> where
+    T: Trait
+{
+	fn create(address: &T::AccountId, amount: T::Balance) -> T::CurrencyId{
+        let token_id = <NextCurrencyId<T>>::get();
+		<NextCurrencyId<T>>::mutate(|id| *id += One::one());
+        let tokens = <Self as MultiTokenCurrency<T::AccountId>>::issue(token_id, amount);
+        // we are creating new token so amount can not be overflowed as its always true
+        // 0 + amount < T::Balance::max_value()
+        let _ = <Self as MultiTokenCurrency<T::AccountId>>::resolve_into_existing(token_id, address, tokens);
+        token_id
+    }
+
+    fn mint(currency_id: T::CurrencyId, address: &T::AccountId, amount: T::Balance) -> DispatchResult {
+        let curren_balance =  <Self as MultiTokenCurrency<T::AccountId>>::total_balance(currency_id, address);
+        // check for overflow while minting
+        let _ = curren_balance.checked_add(&amount).ok_or(Error::<T>::BalanceOverflow)?;
+        let tokens = <Self as MultiTokenCurrency<T::AccountId>>::issue(currency_id, amount);
+        // we are always sure that there will be no overflow as it was checked
+        let _ = <Self as MultiTokenCurrency<T::AccountId>>::resolve_into_existing(currency_id, address, tokens);
```

Comment:
shouldn't this too be resolve_creating instead?

Reply:
ok, i changed it a little bit. I think interface should not allow for minting tokens that does not exists - instead issue should be used. @iStrike7 please take a look on updated impl